crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -49,6 +50,86 @@ impl SystemState {
     }
 }
 
+// TODO: Actually use this in FunctionSystem. We should probably only do this once Systems are constructed using a World reference
+// (to avoid the need for unwrapping to retrieve SystemMeta)
+/// Holds on to persistent state required to drive [`SystemParam`] for a [`System`].  
+pub struct SystemState<Param: SystemParam> {
+    meta: SystemMeta,
+    param_state: <Param as SystemParam>::Fetch,
+    change_tick: u32,
```

Comment:
Small issue: `change_tick` isn't used and `meta.last_change_tick` isn't being set. Gotta sort out the best way to do this.